### PR TITLE
feat(keepldr): Measure the contents of the keep for SEV

### PR DESCRIPTION
**Pull request by haraldh**
*   feat(keepldr): Measure the contents of the keep for SEV
    
    This patch adds a new command `measure`, which produces a measurement on
    the deployed keep and outputs relevant data as a json record,
    which can be used by clients to validate a keep.
    
    This patch implements the measurement for the SEV backend.

### CODE REVIEW

**Comment by haraldh** [comment]
Needs https://github.com/enarx/enarx-keepldr/pull/170 first

**Comment by connorkuehl** [comment]
I have been thinking about something like this too. As of now, our measured initial start is generated at runtime but is a static thing. We don't tune it in any way. We are, I think, wasting our time with generating the image at launch.

I wonder if it would be beneficial to create a tool that can generate a minimal Enarx image with our shim and wasmldr already loaded. Then we can simply measure the image and this measurement can be published, though we should probably GPG-sign it or something. Furthermore, this image can be checked in to the repo. Launching a VM-based keep would involve pasting the image into the address space.

**Comment by haraldh** [comment]
>  Launching a VM-based keep would involve pasting the image into the address space.

~~pasting~~ `mmap`

But yeah... we reached the point, where the final product could just mmap an image for the first mem slot. :partying_face:

**Review by connorkuehl — APPROVED** [review]

File: src/main.rs [hunk 1]
```diff
@@ -115,11 +115,19 @@ struct Exec {
     code: PathBuf,
 }
 
+/// Get a report from a keep
+#[derive(StructOpt)]
+struct Report {
+    /// The payload to run inside the keep
+    code: PathBuf,
+}
```
Comment (lkatalin) [inline src/main.rs:123]: For SGX we've got a `Report` type [here](https://github.com/enarx/sgx/blob/master/src/attestation_types/report.rs#L59) already that conforms to the SGX spec. It seems better to use this directly, unless there is some reason we can't.

**Review by lkatalin — COMMENTED** [review]
See inline.

File: src/main.rs (hunk 1 above)
Comment (connorkuehl) [reply src/main.rs:123]: This is a command-line arg struct. `StructOpt` expands this such that one can do `enarx-keepldr report --code "potato.wasm"`.

Comment (lkatalin) [reply src/main.rs:123]: Ahhh, makes sense. Very good.

**Review by lkatalin — APPROVED** [review]

**Pull request merged**